src/coord/src/coord.rs

```diff
@@ -2716,6 +2705,28 @@ where
         }
     }
 
+    fn optimize_expr(
```

Comment:
Doc comments here would help; I think the problem is folks not putting the right things in the right places, having a clearer explanation of intended use would improve things. For example, my reading of this is that calling with `None` could still be problematic, as it leaves behind `NullaryFunc::MzLogicalTimestamp` and then optimizes the expression. So, if nothing else I'm confused as to what needs to be true about invocations here.

Reply:
Ok, check out the new version. Lots more cleanup and comments.